x-pack/plugins/enterprise_search/public/applications/app_search/components/role_mappings/role_mapping.tsx

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect } from 'react';
+
+import { useParams } from 'react-router-dom';
+
+import { useActions, useValues } from 'kea';
+
+import {
+  EuiButton,
+  EuiCheckbox,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiPageContent,
+  EuiPageContentBody,
+  EuiPageHeader,
+  EuiPanel,
+  EuiRadio,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { FlashMessages } from '../../../shared/flash_messages';
+import { SetAppSearchChrome as SetPageChrome } from '../../../shared/kibana_chrome';
+import { Loading } from '../../../shared/loading';
+import {
+  AttributeSelector,
+  DeleteMappingCallout,
+  RoleSelector,
+} from '../../../shared/role_mapping';
+import { ROLE_MAPPINGS_TITLE } from '../../../shared/role_mapping/constants';
+import { AppLogic } from '../../app_logic';
+import { RoleTypes } from '../../types';
+import { roleHasScopedEngines } from '../../utils/role/has_scoped_engines';
+import { Engine } from '../engine/types';
+
+import {
+  DEV_ROLE_TYPE_DESCRIPTION,
+  EDITOR_ROLE_TYPE_DESCRIPTION,
+  ANALYST_ROLE_TYPE_DESCRIPTION,
+  OWNER_ROLE_TYPE_DESCRIPTION,
+  ADMIN_ROLE_TYPE_DESCRIPTION,
+  ADVANCED_ROLE_SELECTORS_TITLE,
+  ROLE_TITLE,
+  FULL_ENGINE_ACCESS_TITLE,
+  FULL_ENGINE_ACCESS_DESCRIPTION,
+  LIMITED_ENGINE_ACCESS_TITLE,
+  LIMITED_ENGINE_ACCESS_DESCRIPTION,
+  ENGINE_ACCESS_TITLE,
+} from './constants';
+import { RoleMappingsLogic } from './role_mappings_logic';
+
+interface RoleMappingProps {
+  isNew?: boolean;
+}
+
+interface AdvanceRoleType {
+  type: RoleTypes;
+  description: string;
+}
+
+const advancedRoleTypes = [
+  {
+    type: 'dev',
+    description: DEV_ROLE_TYPE_DESCRIPTION,
+  },
+  {
+    type: 'editor',
+    description: EDITOR_ROLE_TYPE_DESCRIPTION,
+  },
+  {
+    type: 'analyst',
+    description: ANALYST_ROLE_TYPE_DESCRIPTION,
+  },
+] as AdvanceRoleType[];
+
+const standardRoleTypes = [
+  {
+    type: 'owner',
+    description: OWNER_ROLE_TYPE_DESCRIPTION,
+  },
+  {
+    type: 'admin',
+    description: ADMIN_ROLE_TYPE_DESCRIPTION,
+  },
+] as AdvanceRoleType[];
```

Comment:
[optional, feel free to skip] it might be nice to move these to the `constants.ts` to keep them together, no worries if you'd rather not

Reply:
Sounds good. Would you prefer changing these to constant case? `standardRoleTypes` -> `STANDARD_ROLE_TYPES`?

Reply:
https://github.com/elastic/kibana/pull/94763/commits/3f528a6c573759005625cf6f59dbe1695a986e54

Reply:
Yea that makes sense! I'm good with it, although I'm also not a huge stickler for full caps constants casing so also good with it if you'd rather leave it as-is